features/step_definitions/functional_tests/functional_tests_steps.rb

```diff
@@ -0,0 +1,58 @@
+Given(/^the test package "([^"]*)" is installed on the controller node$/) do |test_package|
+  control_node.rpm_q(test_package)
+end
+
+Given(/^the package "([^"]*)" is installed on the controller node$/) do |client_package|
+  control_node.rpm_q(client_package)
+end
+
+Given(/^the proper cirros test image has been created$/) do
+  test_image_name = "cirros-test-image-uec"
+
+  if control_node.openstack.image.list.find {|img| img.name == test_image_name }
+    control_node.openstack.image.delete(test_image_name)
+  end
+
+  @test_image = control_node.openstack.image.create(
+    test_image_name,
+    copy_from: "http://clouddata.cloud.suse.de/images/cirros-0.3.3-x86_64-disk.img",
+    container_format: :bare,
+    disk_format: :qcow2,
+    public: true
+  )
+
+  wait_for "Image status set 'active'", max: "60 seconds", sleep: "2 seconds" do
+    image = control_node.openstack.image.show(@test_image.id)
+    break if image.status == "active"
+  end
+end
+
+Then(/^all the functional tests for the package "([^"]*)" pass$/) do |package_name|
+  tests_dir = "/var/lib/#{package_name}-test"
+  package_core_name = package_name.match(/python-(.+)/).captures.first
+  env = {
+    "OS_NOVACLIENT_EXEC_DIR" => "/usr/bin",
```

Comment:
Asking out loud: the code is generic, but we're using a novaclient env var here. Or does this env var work for other clients too?